src/cpp/server/server_cc.cc

```diff
@@ -275,33 +242,144 @@ class Server::SyncRequest GRPC_FINAL : public CompletionQueueTag {
   grpc_completion_queue* cq_;
 };
 
+// Implementation of ThreadManager. Each instance of SyncRequestThreadManager
+// manages a pool of threads that poll for incoming Sync RPCs and call the
+// appropriate RPC handlers
+class Server::SyncRequestThreadManager : public ThreadManager {
```

Comment:
I have a more detailed comment below, but essentially I argue that this should be done via composition and not inheritance since it doesn't have the smell of inheritance at all.